fw/cache.c

```diff
@@ -3159,7 +3189,8 @@ tfw_cache_start(void)
 		goto close_db;
 	}
 #endif
-	tfw_init_node_cpus();
+	if(tfw_init_node_cpus() == -ENOMEM) 
+		goto close_db;
```

Comment:
In case of error you're leaking memory. You need to add a new label to free the allocated per-node CPU maps.